src/Symfony/Bundle/TwigBundle/ContainerAwareRuntimeLoader.php

```diff
@@ -22,23 +22,32 @@
 class ContainerAwareRuntimeLoader implements \Twig_RuntimeLoaderInterface
 {
     private $container;
-    private $mapping;
     private $logger;
 
-    public function __construct(ContainerInterface $container, array $mapping, LoggerInterface $logger = null)
+    public function __construct(ContainerInterface $container, $logger = null)
     {
-        $this->container = $container;
-        $this->mapping = $mapping;
+        //  BC 3.x, to be removed in 4.0, re-adding the LoggerInterface typehint to the $logger argument
+        if (is_array($logger)) {
+            @trigger_error(sprintf('The second argument of %s() is deprecated since version 3.3 and will be removed, passing something else than a %s instance will trigger an error in 4.0.', __METHOD__, LoggerInterface::class), E_USER_DEPRECATED);
+
+            if (2 < func_num_args() && func_get_arg(3) instanceof LoggerInterface) {
```

Comment:
not needed if the constructor signature is changed to 
`__construct(ContainerInterface $container, $mapping = null, LoggerInterface $logger = null)`